src/app.js

```
@@ -1 +1,58 @@
-// write code here
+/* eslint-disable no-console */
+const fs = require('fs');
+const path = require('path');
+
+function move() {
+  if (process.argv.length < 4) {
+    console.error(
+      'Error: need 4 args: node src/app.js sourcePath destinationPath',
+    );
+
+    return;
+  }
+
+  const [file, destination] = process.argv.slice(2);
+
+  if (file === destination) {
+    return;
+  }
+
+  if (!fs.existsSync(file)) {
+    console.error('Error: the source file does not exist');
+
+    return;
+  }
+
+  const destExists = fs.existsSync(destination);
+  let isDestDirectory = false;
+
+  if (!destExists && destination[destination.length - 1] === '/') {
+    console.error('Error: the destination file does not exist');
+  }
+
+  if (destExists) {
+    isDestDirectory = fs.statSync(destination).isDirectory();
+  }
+
+  let targetPath = destination;
+
+  if (isDestDirectory) {
+    const fileName = path.basename(file);
+
+    targetPath = path.join(destination, fileName);
+  } else if (destination.endsWith('/')) {
+    fs.mkdirSync(destination, { recursive: true });
```

According to the task description, 'In case destination directory does not exist an error must be thrown.' Instead of creating the directory, you should throw an error here.